### ISSUE TITLE
--self init-only directive causes error: Unexpected class/static func with static class function

### ISSUE THREAD

**Issue by tkafka** [nicklockwood/SwiftFormat#975]
Here is a minimal reproduction test case of a valid swift code (shortened from a real use case where the problem manifests for me): https://gist.github.com/tkafka/f7914dcca74770161986cfda65c4951d

After removing the `--self init-only` directive from `.swiftformat`, swiftformat works ok. However, I found initializers without self (and with parameters similarly named to class members) confusing, so I'd love if this could be fixed. Thank you!

```
swiftformat .
Running SwiftFormat...
Reading config file at /Users/kafkat/Dev/swiftformat-problem/.swiftformat
error: Unexpected class/static func on line 10 in /Users/kafkat/Dev/swiftformat-problem/Test.swift.
```

I use `swiftformat 0.48.9` from homebrew on M1 mac.

**Comment by nicklockwood** [comment]
Awesome, thanks 👍

**Comment by tkafka** [comment]
Thank you! It seems that removing the `guard` in `aaa()` causes the `bbb()` to not cause the error anymore, hope this might help with debugging.

**Comment by nicklockwood** [comment]
@tkafka fixed in 0.48.10

**Comment by tkafka** [comment]
@nicklockwood Thank you! I can confirm on a full project as well!
